### PR TITLE
fix race (out of order) in flushing topics (wrong state is stored in changelog)

### PR DESCRIPTION
**Pull request by trauter**
## Description

This PR fixes a race condition in writing messages to topics that resulted in a violation of the ordering guarantee (especially changelog topics).

What happened before:
The producer buffer is filled with messages. 
_handle_pending is taking one message from the buffer at a time (Participant 1).
If the buffer grows flush_atmost is called which itself starts taking messages from the buffer and sends them (Participant 2).

In rare cases, both participants popped message (e.g. MSG1, then MSG2), but the messages are written in reverse order (e.g. MSG2, then MSG1) to the topic.
In case of changelog topics, this results in wrong states if both messages have the same key.

The fix:
Basically, flush simply waits until the buffer is empty.

### CODE REVIEW

**Comment by trauter** [comment]
have to fix tests before..
